src/components/lookup-container/index.js

```diff
@@ -20,6 +21,22 @@ class LookupContainer extends Component {
 
 	inputs = [];
 
+	componentDidMount() {
+		const methodParam = getParam( 'method' );
+		if ( methodParam ) {
+			this.props.updateMethod( methodParam );
+		}
+
+		const pathParam = getParam( 'path' );
+		if ( pathParam ) {
+			this.props.updateUrl( pathParam );
+		}
+	}
```

Comment:
instead of dispatching actions to change the URL after loading, we should be able to set the default store values in `src/state/request/reducer.js` so that it's properly set at startup.

since the store code might not exist within a browser environment it would be wise to default to the current empty defaults if `window.location` doesn't exist.

```js
// state/request/reducer.js

const defaultStateFromUrl = () => {
	if ( 'undefined' === typeof window || ! window.location ) {
		return {};
	}

	const searchParams = new URLSearchParams( window.location.search );
	return Object.assign(
		{} ,
		searchParams.has( 'method' ) && { method: searchParams.get( 'method' ) },
		searchParams.has( 'path' ) && { url: searchParams.get( 'url' ) },
	);
}

const defaultState = Object.assign(
	{
		method: 'GET',
		endpoint: false,
		pathValues: {},
		url: '',
		queryParams: {},
		bodyParams: {},
	},
	defaultStateFromUrl()
);
```

of course, if we wanted further validation that would also make sense.

```js
const parseMethod = ( method ) => {
	if ( ! method || 'string' !== typeof method ) {
		return;
	}

	switch ( method.toUpperCase() ) {
		case 'GET':
		case 'POST':
		case 'PUT':
		case 'DELETE':
		case 'PATCH':
			return method.toUpperCase();
	}
}

const parsePath = ( path ) => {
	try {
		const url = new URL( path, 'https://example.com' ) );
		return `${ url.pathname }${ url.search }${ url.hash }`;
	} catch ( e ) {
		return;
	}
}

const defaultStateFromUrl = () => {
	if ( 'undefined' === typeof window || ! window.location ) {
		return {};
	}

	const searchParams = new URLSearchParams( window.location.search );

	const method = parseMethod( searchParams.get( 'method' ) );
	const url = parsePath( searchParams.get( 'path' ) );

	return Object.assign(
		{},
		method && { method },
		url && { url },
	);
}
```